backends/magma/magma_interp.c

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+#include "ceed-magma.h"
+
+//////////////////////////////////////////////////////////////////////////////////////////
+#ifdef __cplusplus
+CEED_INTERN "C"
+#endif
+magma_int_t
+magma_interp(
+  magma_int_t P, magma_int_t Q,
+  magma_int_t dim, magma_int_t ncomp,
+  const CeedScalar *dT, CeedTransposeMode tmode,
+  const CeedScalar *dU, magma_int_t u_elstride, magma_int_t u_compstride,
+  CeedScalar *dV, magma_int_t v_elstride, magma_int_t v_compstride,
+  magma_int_t nelem, magma_kernel_mode_t kernel_mode, magma_int_t *maxthreads,
+  magma_queue_t queue) {
+  magma_int_t launch_failed = 0;
+
+  if(kernel_mode == MAGMA_KERNEL_DIM_SPECIFIC) {
+    switch(dim) {
+    case 1: launch_failed = magma_interp_1d(P, Q, ncomp, dT, tmode, dU, u_elstride,
+                                              u_compstride, dV, v_elstride, v_compstride, nelem, maxthreads[0], queue); break;
+    case 2: launch_failed = magma_interp_2d(P, Q, ncomp, dT, tmode, dU, u_elstride,
+                                              u_compstride, dV, v_elstride, v_compstride, nelem, maxthreads[1], queue); break;
+    case 3: launch_failed = magma_interp_3d(P, Q, ncomp, dT, tmode, dU, u_elstride,
+                                              u_compstride, dV, v_elstride, v_compstride, nelem, maxthreads[2], queue); break;
+    default: launch_failed = 1;
```

Comment:
The style here looks strange. If you put the `launch_failed = ...` on a new line, then `make style` will better align the contents after a line break. I'm also not sure why `make style` isn't inserting a line break when the contents go out to column 127.